gc.c

```diff
@@ -1861,9 +1861,40 @@ static const rb_data_type_t id_to_obj_tbl_type = {
     .flags = RUBY_TYPED_WB_PROTECTED | RUBY_TYPED_FREE_IMMEDIATELY
 };
 
+static VALUE
+class_object_id(VALUE klass)
+{
+    VALUE id = RCLASS(klass)->object_id;
```

Comment:
It might be nice to do an atomic load here, otherwise TSan will complain, but I think in practice it doesn't matter here (ie. an atomic load with memory order relaxed should work). So not a blocker.

Reply:
Actually, I couldn't make it work, so I used the same hack you used in `string.c`. Sorry.